src/oic/oauth2/provider.py

```diff
@@ -1115,14 +1115,18 @@ def verify_endpoint(self, request="", cookie=None, **kwargs):
         kwargs["cookie"] = cookie
         return authn.verify(_req, **kwargs)
 
-    def write_session_cookie(self, value):
-        return make_cookie(self.session_cookie_name, value, self.seed, path="/")
+    def write_session_cookie(self, value, http_only=True):
+        return make_cookie(self.session_cookie_name, value, self.seed, path="/", httponly=http_only)
```

Comment:
If IFrames are the usecase, we might want to handle SameSite settings for cookies as well. 
Otherwise it will probably break with newer browsers like Chrome 80 that implement SameSite=Lax by default.

(https://chromestatus.com/features#samesite), 
https://textslashplain.com/2019/09/30/same-site-cookies-by-default/

Reply:
IFrames is the use case. I'll look at SameSite. Any preference as how to handle it ?

Reply:
The make_cookie() method should handle and explitly set a SameSite value. This may requires monkey patching the Cookie class of Python (see Beaker/Cookie class), as it is only supported in 3.8+.

The patch is like this:
`
from http_cookies import Morsel
Morsel._reserved[str('samesite')] = str('SameSite')
`

In IFrames usage the only value for SameSite that works properly is 'None'.
https://web.dev/samesite-cookie-recipes/
or mentioning it might break OpenID connect logins:
https://devblogs.microsoft.com/aspnet/upcoming-samesite-cookie-changes-in-asp-net-and-asp-net-core/